insonmnia/hub/server.go

```diff
@@ -227,6 +275,17 @@ func (h *Hub) applyFilters(miner *MinerCtx, req *pb.TaskRequirements) (bool, err
 
 // StartTask schedules the Task on some miner
 func (h *Hub) StartTask(ctx context.Context, request *pb.HubStartTaskRequest) (*pb.HubStartTaskReply, error) {
+	if !h.isLeader {
```

Comment:
Seems like boilerplate. Maybe rewrite using reflection to simple forward the request without looking into it?